include/wally_psbt.h

```diff
@@ -0,0 +1,429 @@
+ *
+ * :param redeem_script: The redeem script needed for spending this output
+ * :param redeem_script_len: The length of the redeem script.
+ * :param witness_script: The witness script needed for spending for this output
+ * :param witness_script_len: The length of the witness script.
+ * :param keypaths: The HD keypaths for the keys needed for spending this output
+ * :param unknowns: The unknown key value pairs for this output.
+ * :param output: Destination for the resulting psbt output.
+ */
+WALLY_CORE_API int wally_psbt_output_init_alloc(
+    unsigned char *redeem_script,
+    size_t redeem_script_len,
+    unsigned char *witness_script,
+    size_t witness_script_len,
+    struct keypath_map *keypaths,
+    struct unknowns_map *unknowns,
+    struct wally_psbt_output **output);
+
+/**
+ * Add or replace items in an output
+ *
+ * :param input: The output to update
+ * :param redeem_script: The redeem script needed for spending this output
+ * :param redeem_script_len: The length of the redeem script.
+ * :param witness_script: The witness script needed for spending for this output
+ * :param witness_script_len: The length of the witness script.
+ * :param keypaths: The HD keypaths for the keys needed for spending this output
+ * :param unknowns: The unknown key value pairs for ths output.
+ */
+WALLY_CORE_API int wally_psbt_output_update(
+    struct wally_psbt_output *output,
+    unsigned char *redeem_script,
+    size_t redeem_script_len,
+    unsigned char *witness_script,
+    size_t witness_script_len,
+    struct keypath_map *keypaths,
+    struct unknowns_map *unknowns);
+
+#ifndef SWIG_PYTHON
+/**
+ * Free a psbt input allocated by `wally_psbt_output_init_alloc`.
+ *
+ * :param input: The psbt input to free.
+ */
+WALLY_CORE_API int wally_psbt_output_free(struct wally_psbt_output *output);
+#endif /* SWIG_PYTHON */
+
+/**
+ * Allocate and initialize a new psbt.
+ *
+ * :param inputs_allocation_len: The number of inputs to pre-allocate space for.
+ * :param outputs_allocation_len: The number of outputs to pre-allocate space for.
+ * :param global_unknowns_allocation_len: The number of global unknowns to allocate space for.
+ * :param output: Destination for the resulting psbt output.
+ */
+WALLY_CORE_API int wally_psbt_init_alloc(
+    size_t inputs_allocation_len,
+    size_t outputs_allocation_len,
+    size_t global_unknowns_allocation_len,
+    struct wally_psbt **output);
+
+#ifndef SWIG_PYTHON
+/**
+ * Free a psbt allocated by `wally_psbt_init_alloc`.
+ *
+ * :param psbt: The psbt to free.
+ */
+WALLY_CORE_API int wally_psbt_free(struct wally_psbt *psbt);
+#endif /* SWIG_PYTHON */
+
+/**
+ * Set the global transaction for a psbt.
+ * Also initializes all of the wally_psbt_input and wally_psbt_outputs necessary
+ *
+ * :param tx: The transaction to set.
+ * :param psbt: The psbt to set the transaction for
+ */
+WALLY_CORE_API int wally_psbt_set_global_tx(
+    struct wally_tx *tx,
+    struct wally_psbt *psbt);
```

Comment:
for consistency it'd appear the arguments should be psbt, tx. appears reversed as is.

Reply:
Done